test/metabase/db/custom_migrations_test.clj

```diff
@@ -1014,3 +1020,445 @@
+                                      {:click_behavior
+                                       {:type             "link",
+                                        :linkType         "url",
+                                        :linkTemplate     "http://example.com/{{ID}}",
+                                        :linkTextTemplate "here's an id: {{ID}}"}},
+                                      "[\"ref\",[\"field\",6,null]]"
+                                      {:click_behavior
+                                       {:type             "link",
+                                        :linkType         "url",
+                                        :linkTemplate     "http://example.com//{{id}}",
+                                        :linkTextTemplate "here is my id: {{id}}"}}}}]
+              (f)
+              (is (= expected-settings
+                     (-> (t2/select-one DashboardCard :id dashcard-id)
+                         :visualization_settings))))))]
+    (testing "Running the migration from scratch"
+      (impl/test-migrations ["v48.00-022"] [migrate!]
+        (expect-correct-settings! migrate!)))
+    (testing "Running the migration after a previous data-migration still works"
+      (impl/test-migrations ["v48.00-022"] [migrate!]
+        (expect-correct-settings! (fn []
+                                    (#'custom-migrations/migrate-click-through!)
+                                    (migrate!)))))))
+
+(defn- get-json-setting
+  [setting-k]
+  (json/parse-string (t2/select-one-fn :value setting/Setting :key (name setting-k))))
+
+(defn- call-with-ldap-and-sso-configured [ldap-group-mappings sso-group-mappings f]
+  (mt/with-temporary-raw-setting-values
+    [ldap-group-mappings    (json/generate-string ldap-group-mappings)
+     saml-group-mappings    (json/generate-string sso-group-mappings)
+     jwt-group-mappings     (json/generate-string sso-group-mappings)
+     saml-enabled           "true"
+     ldap-enabled           "true"
+     jwt-enabled            "true"]
+    (f)))
+
+(defmacro ^:private with-ldap-and-sso-configured
+  "Run body with ldap and SSO configured, in which SSO will only be configured if enterprise is available"
+  [ldap-group-mappings sso-group-mappings & body]
+  (binding [setting/*allow-retired-setting-names* true]
+    `(call-with-ldap-and-sso-configured ~ldap-group-mappings ~sso-group-mappings (fn [] ~@body))))
+
+;; The `remove-admin-from-group-mapping-if-needed` migration is written to run in OSS version
+;; even though it might make changes to some enterprise-only settings.
+;; In order to write tests that runs in both OSS and EE, we can't use
+;; [[metabase.models.setting/get]] and [[metabase.test.util/with-temporary-setting-values]]
+;; because they require all settings are defined.
+;; That's why we use a set of helper functions that get setting directly from DB during tests
+(deftest migrate-remove-admin-from-group-mapping-if-needed-test
```

Comment:
nit: on optimizing test speed time, I wonder if we should consider this as `behavior` test just call `migrate-remove-admin-from-group-mapping-if-needed` directly. instead of doing the setup with `test-migrations` to not have to set up the DB 4 times here.

And instead, we should write a test for upgrading scenarios like: if already executed, this migration should be skipped etc.

This is just a suggestion, feel free to skip it if it doesn't make much sense to you.